src/resolve/resolved-conf.c

```diff
@@ -228,6 +232,149 @@ int config_parse_search_domains(
         return 0;
 }
 
+int config_parse_dnssd_service_name(const char *unit, const char *filename, unsigned line, const char *section, unsigned section_line, const char *lvalue, int ltype, const char *rvalue, void *data, void *userdata) {
+        static const Specifier specifier_table[] = {
+                { 'h', specifier_host_name, NULL },
```

Comment:
i'd probably hook up specifier `specifier_machine_id`, `specifier_boot_id`, `specifier_kernel_release` here too. Not because they are necessarily the most useful in this context, but simply to minimize confusion, since we support them in units and tmpfiles too, and it would be surprising if we didn't here...